### PR TITLE
T22380 Add support for Hold() and Release() D-Bus methods

### PR DESCRIPTION
**Pull request by pwithnall**
https://phabricator.endlessm.com/T22380

### CODE REVIEW

File: libmogwai-schedule/schedule-service.c [hunk 1]
```diff
@@ -1472,6 +1544,156 @@ schedule_cb (GObject      *obj,
...
+{
+  MwsPeerManager *peer_manager = MWS_PEER_MANAGER (obj);
+  g_autoptr(HoldData) data = user_data;
+  MwsScheduleService *self = data->schedule_service;
+  GDBusMethodInvocation *invocation = data->invocation;
+  g_autoptr(GError) local_error = NULL;
+
+  /* Finish looking up the sender. */
+  g_autofree gchar *sender_path = NULL;
+  sender_path = mws_peer_manager_ensure_peer_credentials_finish (peer_manager,
```
Comment (joaquimrocha) [inline libmogwai-schedule/schedule-service.c]: I'd just break the line after the `=`...

File: mogwai-schedule-client/main.c [hunk 2]
```diff
@@ -609,6 +609,232 @@ handle_download (RunContext  *run_context,
...
+      { "bus-address", 'a', G_OPTION_FLAG_NONE, G_OPTION_ARG_STRING, &bus_address,
+        N_("Address of the D-Bus daemon to connect to (default: system bus)"),
+        N_("ADDRESS") },
+      { "quiet", 'q', G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE, &quiet,
+        N_("Only print error messages and signal notifications"), NULL },
+      { NULL, },
+    };
+
+  g_autoptr(GOptionContext) context = NULL;
+  context = g_option_context_new (NULL);
```
Comment (joaquimrocha) [inline mogwai-schedule-client/main.c]: No need for a newline here AFAICS.

File: mogwai-schedule-client/main.c [hunk 3]
```diff
@@ -609,6 +609,232 @@ handle_download (RunContext  *run_context,
...
+  g_option_context_set_summary (context, _("Monitor download scheduler properties"));
+  g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
+
+  if (!g_option_context_parse (context, argc, argv, &error))
+    {
+      g_autofree gchar *message = NULL;
+      message = g_strdup_printf (_("Option parsing failed: %s"),
+                                 error->message);
```
Comment (joaquimrocha) [inline mogwai-schedule-client/main.c]: No need for a newline AFAICS.

File: mogwai-schedule-client/main.c [hunk 4]
```diff
@@ -609,6 +609,232 @@ handle_download (RunContext  *run_context,
...
+  gboolean invalidated = FALSE;
+
+  g_signal_connect (scheduler, "invalidated",
+                    (GCallback) invalidated_cb, &invalidated);
+  g_signal_connect (scheduler, "notify", (GCallback) notify_cb, NULL);
+
+  g_autoptr(GSource) cancellable_source = NULL;
+  cancellable_source = g_cancellable_source_new (run_context->cancellable);
+  g_source_set_callback (cancellable_source, cancelled_cb, NULL, NULL);
+  g_source_attach (cancellable_source, NULL);
```
Comment (joaquimrocha) [inline mogwai-schedule-client/main.c]: :+1:

File: libmogwai-schedule/schedule-service.c (hunk 1 above)
Comment (pwithnall) [reply libmogwai-schedule/schedule-service.c]: Meh, results in a slightly longer line.

File: mogwai-schedule-client/main.c (hunk 3 above)
Comment (pwithnall) [reply mogwai-schedule-client/main.c]: There is, by 1 character.

**Comment by pwithnall** [comment]
Rebased, one comment addressed, and merged. Thanks for the review.

**Pull request closed without merging**